src/main/java/org/skife/jdbi/v2/MappingRegistry.java

```diff
@@ -49,32 +58,75 @@ public void add(ResultSetMapper mapper)
 
     public void add(ResultSetMapperFactory factory)
     {
-        factories.add(factory);
-        cache.clear();
+        rowFactories.add(factory);
+        rowCache.clear();
     }
 
     public ResultSetMapper mapperFor(Class type, StatementContext ctx) {
-        if (cache.containsKey(type)) {
-            ResultSetMapper mapper = cache.get(type);
+        if (rowCache.containsKey(type)) {
+            ResultSetMapper mapper = rowCache.get(type);
             if (mapper != null) {
                 return mapper;
             }
         }
 
-        for (ResultSetMapperFactory factory : factories) {
+        for (ResultSetMapperFactory factory : rowFactories) {
             if (factory.accepts(type, ctx)) {
                 ResultSetMapper mapper =  factory.mapperFor(type, ctx);
-                cache.put(type, mapper);
+                rowCache.put(type, mapper);
                 return mapper;
             }
         }
 
+        // TODO remove this and let the column mapper block below take over
         if (BUILT_IN_MAPPERS.accepts(type, ctx)) {
             ResultSetMapper mapper = BUILT_IN_MAPPERS.mapperFor(type, ctx);
-            cache.put(type, mapper);
+            rowCache.put(type, mapper);
+            return mapper;
+        }
+
+        ResultColumnMapper columnMapper = columnMapperFor(type, ctx);
+        if (columnMapper != null) {
+            ResultSetMapper mapper = new SingleColumnMapper(columnMapper);
+            rowCache.put(type, mapper);
             return mapper;
         }
 
         throw new DBIException("No mapper registered for " + type.getName()) {};
     }
+
+    public void addColumn(ResultColumnMapper mapper)
+    {
+        this.addColumn(new InferredColumnMapperFactory(mapper));
+    }
+
+    public void addColumn(ResultColumnMapperFactory factory) {
+        columnFactories.add(factory);
+        columnCache.clear();
+    }
+
+    public ResultColumnMapper columnMapperFor(Class type, StatementContext ctx) {
+        if (columnCache.containsKey(type)) {
+            ResultColumnMapper mapper = columnCache.get(type);
+            if (mapper != null) {
+                return mapper;
+            }
+        }
+
+        for (ResultColumnMapperFactory factory : columnFactories) {
+            if (factory.accepts(type, ctx)) {
+                ResultColumnMapper mapper =  factory.columnMapperFor(type, ctx);
+                columnCache.put(type, mapper);
+                return mapper;
+            }
+        }
+
+        if (BUILD_IN_COLUMN_MAPPERS.accepts(type, ctx)) {
```

Comment:
This matches the existing design of the `ResultSetMapper` registry, and I think we should stick with that pattern.

Mappers and mapper factories are being consulted in the order they were registered. Built in mappers are consulted last. This allows the user to substitute their own mapper for built in types, if they choose.

If we were to eagerly add the built in mappers, they would be at the front of the line, before any user-registered mappers, which would take away the ability to provide substitute mappers.